### PR TITLE
Consumer groups pagination (Backend)

### PR DESCRIPTION
**Pull request by iliax**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing applications:)

**What changes did you make?** (Give an overview)

* consumer-groups endpoint with pagination added
* consumer group dto mappings moved to controller
* consumer groups loading logic refactored

**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "X" next to an item)

- [ ] No need to
- [X] Manually(please, describe, when necessary)
- [X] Unit checks
- [X] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "X" next to an item, otherwise PR will fail)
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [X] My changes generate no new warnings(e.g. Sonar is happy)
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

### CODE REVIEW

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ConsumerGroupService.java [hunk 1]
```diff
@@ -33,88 +35,145 @@
...
-            .flatMap(descriptions ->
-                Flux.fromIterable(descriptions)
-                    .parallel()
-                    .flatMap(d ->
-                        ac.listConsumerGroupOffsets(d.groupId())
-                            .map(offsets -> ClusterUtil.convertToInternalConsumerGroup(d, offsets))
-                    )
-                    .sequential()
-                    .collectList()));
+  @Deprecated // need to migrate to pagination
```
Comment (Haarolean) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ConsumerGroupService.java:65]: Drop deprecated methods in next release once frontend stops using the old ones

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ReactiveAdminClient.java [hunk 2]
```diff
@@ -293,9 +294,18 @@ public String getVersion() {
...
 
-  public Mono<Map<TopicPartition, OffsetAndMetadata>> listConsumerGroupOffsets(String groupId) {
-    return toMono(client.listConsumerGroupOffsets(groupId).partitionsToOffsetAndMetadata())
-        .map(MapUtil::removeNullValues);
+  public Mono<Map<TopicPartition, Long>> listConsumerGroupOffsets(String groupId) {
+    return listConsumerGroupOffsets(groupId, null);
+  }
+
+  public Mono<Map<TopicPartition, Long>> listConsumerGroupOffsets(
+      String groupId, @Nullable List<TopicPartition> partitions) {
```
Comment (Haarolean) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ReactiveAdminClient.java:302]: I'd prefer an empty collection over a nullable one -- less possible NPEs.

**Review by Haarolean — CHANGES_REQUESTED** [review]

**Review by Haarolean — APPROVED** [review]
LGTM, but maybe @germanosin could take a look additionally

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/ConsumerGroupsController.java [hunk 3]
```diff
@@ -33,36 +45,157 @@
...
+    }
+    details.setPartitions(new ArrayList<>(partitionMap.values()));
+    return details;
+  }
+
+  private static BrokerDTO mapCoordinator(Node node) {
+    return new BrokerDTO().host(node.host()).id(node.id());
+  }
+
+  private static ConsumerGroupStateDTO mapConsumerGroupState(
```
Comment (germanosin) [inline kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/ConsumerGroupsController.java:188]: Maybe mapping class would be better for this?

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml [hunk 4]
```diff
@@ -615,6 +615,46 @@ paths:
                 items:
                   $ref: '#/components/schemas/ConsumerGroup'
 
+  /api/clusters/{clusterName}/consumer-groups/paged:
```
Comment (germanosin) [inline kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:618]: Why do not use the same url with old-style/ new style behavior?

**Review by germanosin — CHANGES_REQUESTED** [review]

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/ConsumerGroupsController.java [hunk 5]
```diff
@@ -33,36 +45,157 @@
...
+      String clusterName,
+      Integer page,
+      Integer perPage,
+      String search,
+      ConsumerGroupOrderingDTO orderBy,
+      ServerWebExchange exchange) {
+    return consumerGroupService.getConsumerGroupsPage(
+            getCluster(clusterName),
+            Optional.ofNullable(page).filter(i -> i > 0).orElse(1),
+            Optional.ofNullable(perPage).filter(i -> i > 0).orElse(25),
```
[the remark this answers is not in the document]
Comment (iliax) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/ConsumerGroupsController.java]: done

File: kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/ConsumerGroupsController.java (hunk 3 above)
Comment (iliax) [reply kafka-ui-api/src/main/java/com/provectus/kafka/ui/controller/ConsumerGroupsController.java]: done

File: kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml (hunk 4 above)
Comment (iliax) [reply kafka-ui-contract/src/main/resources/swagger/kafka-ui-api.yaml:618]: in paging endpont we need to return total number of pages, so return types are incompatible

**Comment by iliax** [comment]
@germanosin pls recheck

**Review by germanosin — APPROVED** [review]

**Pull request merged**